jgrapht-core/src/main/java/org/jgrapht/alg/isomorphism/AHUForestIsomorphismInspector.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.isomorphism;
+
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+import org.jgrapht.alg.connectivity.ConnectivityInspector;
+import org.jgrapht.graph.SimpleGraph;
+
+import java.util.*;
+import java.util.function.Supplier;
+
+/**
+ * This is an implementation of the AHU algorithm for detecting an isomorphism between two rooted forests.
+ * Please see <a href="http://mathworld.wolfram.com/GraphIsomorphism.html">mathworld.wolfram.com</a> for a complete
+ * definition of the isomorphism problem for general graphs.
+ *
+ * <p>
+ *     The original algorithm was first presented in "Alfred V. Aho and John E. Hopcroft. 1974.
+ *     The Design and Analysis of Computer Algorithms (1st ed.). Addison-Wesley Longman Publishing Co., Inc., Boston, MA, USA."
+ * </p>
+ *
+ * <p>
+ *     This implementation runs in linear time (in the number of vertices of the input forests)
+ *     while using a linear amount of memory.
+ * </p>
+ *
+ * <p>
+ *      For an implementation that supports both unrooted and rooted trees see {@link AHUTreeIsomorphismInspector}.
+ * </p>
+ *
+ * <p>
+ *     Note: This implementation requires the input graphs to be modifiable and to have valid vertex suppliers
+ *     (see {@link Graph#getVertexSupplier()}).
+ * </p>
+ *
+ * @param <V> the type of the vertices
+ * @param <E> the type of the edges
+ *
+ * @author Alexandru Valeanu
+ */
+public class AHUForestIsomorphismInspector<V, E> {
```

Comment:
Why doesn't this implement the IsomorphismInspector interface?  It's OK if you only return a single isomorphism instead of all possible ones; just document that.

Reply:
"This inspector only returns a single mapping (chosen arbitrarily) rather than all possible mappings."